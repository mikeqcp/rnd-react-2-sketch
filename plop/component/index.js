const path = require('path');

const templatesPath = path.join(__dirname, 'templates');

module.exports = (plop) => {
  const containerDirectory = 'src/{{ directory }}/{{ camelCase name }}';
  plop.setGenerator('simple component', {
    description: 'Simple component',
    prompts: [{
      type: 'input',
      name: 'name',
      message: 'Name:',
    }, {
      type: 'directory',
      name: 'directory',
      basePath: 'src',
      message: 'Component\'s directory:',
    }],
    actions: [{
      type: 'add',
      path: `${containerDirectory}/{{ camelCase name }}.component.js`,
      templateFile: path.join(templatesPath, 'component.hbs'),
    }, {
      type: 'add',
      path: `${containerDirectory}/{{ camelCase name }}.styles.js`,
      templateFile: path.join(templatesPath, 'styles.hbs'),
    }, {
      type: 'add',
      path: `${containerDirectory}/index.js`,
      templateFile: path.join(templatesPath, 'index.hbs'),
    }],
  });
};
