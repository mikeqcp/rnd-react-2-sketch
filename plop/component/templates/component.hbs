import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Container } from './{{ camelCase name }}.styles';

export class {{ pascalCase name }} extends PureComponent {
  render() {
    return (
      <Container>
        {{ pascalCase name }} component
      </Container>
    );
  }
}

{{ pascalCase name }}.propTypes = {
};
