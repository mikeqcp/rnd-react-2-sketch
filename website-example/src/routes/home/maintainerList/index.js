export { MaintainerList } from './maintainerList.component';
