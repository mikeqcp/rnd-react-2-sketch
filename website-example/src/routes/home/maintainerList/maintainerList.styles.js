import styled from 'styled-components';


export const Container = styled.div`
 display: flex;
 flex-direction: column;
 align-items: center;
 justify-content: center;
 text-align: center;
`;

export const List = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0 0 20px;
`;

export const ListItem = styled.li`
  margin: 0;
`;
