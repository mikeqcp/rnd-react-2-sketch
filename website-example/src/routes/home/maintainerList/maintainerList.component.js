import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl } from 'react-intl';
import { compose } from 'ramda';

import messages from './maintainersList.messages';
import { Container, List, ListItem } from './maintainerList.styles';
import { H2, Link } from '../../../theme/typography';


export class MaintainerListComponent extends PureComponent {
  static propTypes = {
    items: PropTypes.object.isRequired,
  };

  render() {
    return (
      <Container>
        <H2>
          <FormattedMessage {...messages.title} />
        </H2>

        <List>
          {this.props.items.toArray().map((item, key) => (
            <ListItem key={key}>
              <Link href={`mailto:${item.get('email')}`}>
                {item.get('firstName')} {item.get('lastName')}
              </Link>
            </ListItem>
          ))}
        </List>
      </Container>
    );
  }
}

export const MaintainerList = compose(
  injectIntl,
)(MaintainerListComponent);
