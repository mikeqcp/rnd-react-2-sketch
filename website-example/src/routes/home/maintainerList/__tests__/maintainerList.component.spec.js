import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { MaintainerListComponent as MaintainerList } from '../maintainerList.component';


describe('MaintainerList: Component', () => {
  const defaultProps = {
    items: fromJS([
      { email: 'email1@domain.com', lastName: 'maintainer1' },
      { email: 'email2@domain.com', lastName: 'maintainer2' },
    ]),
  };

  const component = (props) => (
    <MaintainerList {...defaultProps} {...props} />
  );

  it('should render correctly', () => {
    const wrapper = shallow(component());
    global.expect(wrapper).toMatchSnapshot();
  });
});
