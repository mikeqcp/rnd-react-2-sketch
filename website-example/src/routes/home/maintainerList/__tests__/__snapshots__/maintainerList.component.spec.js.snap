// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaintainerList: Component should render correctly 1`] = `
<styled.div>
  <styled.h2>
    <FormattedMessage
      defaultMessage="Maintainers"
      id="home.maintainerList.title"
      values={Object {}}
    />
  </styled.h2>
  <styled.ul>
    <styled.li
      key="0"
    >
      <styled.a
        href="mailto:email1@domain.com"
      >
         
        maintainer1
      </styled.a>
    </styled.li>
    <styled.li
      key="1"
    >
      <styled.a
        href="mailto:email2@domain.com"
      >
         
        maintainer2
      </styled.a>
    </styled.li>
  </styled.ul>
</styled.div>
`;
